internal/testrunner/runners/pipeline/test_result.go

```diff
@@ -103,24 +104,37 @@ func compareJsonNumbers(a, b json.Number) bool {
 }
 
 func diffJson(want, got []byte) (string, error) {
-	opts := jsondiff.DefaultConsoleOptions()
-
-	// Remove colored output.
-	opts.Added = jsondiff.Tag{Begin: "+ "}
-	opts.Removed = jsondiff.Tag{Begin: "- "}
-	opts.Changed = jsondiff.Tag{}
-	opts.Skipped = jsondiff.Tag{}
-
-	// Configure diff.
-	opts.SkipMatches = true
-	opts.CompareNumbers = compareJsonNumbers
+	var gotVal, wantVal interface{}
+	err := jsonUnmarshalUsingNumber(want, &wantVal)
+	if err != nil {
+		return "", fmt.Errorf("invalid want value: %w", err)
+	}
+	err = jsonUnmarshalUsingNumber(got, &gotVal)
+	if err != nil {
+		return "", fmt.Errorf("invalid got value: %w", err)
+	}
+	if cmp.Equal(gotVal, wantVal, cmp.Comparer(compareJsonNumbers)) {
+		return "", nil
+	}
 
-	result, diff := jsondiff.Compare(want, got, &opts)
-	switch result {
-	case jsondiff.FirstArgIsInvalidJson, jsondiff.SecondArgIsInvalidJson, jsondiff.BothArgsAreInvalidJson:
-		return "", errors.New("invalid json")
+	got, err = marshalNormalizedJSON(gotVal)
+	if err != nil {
+		return "", err
 	}
-	return diff, nil
+	want, err = marshalNormalizedJSON(wantVal)
```

Comment:
`marshalNormalizedJSON` marshals and unmarshals using `jsonUnmarshalUsingNumber`, something that has been already done here. We could do the final marshal directly here instead of repeating the operations.
Not so important though as this code is only used if the documents are different.

Reply:
I was being conservative to ensure that the values were being treated the same way and to ensure that the reader was clear that this was happening.